public/app/features/explore/state/explorePane.ts

```diff
@@ -76,6 +77,52 @@ export function changePanelState(
   };
 }
 
+/**
+ * Tracks the state of correlation helper data in the panel
+ */
+interface ChangeCorrelationHelperData {
+  exploreId: string;
+  correlationEditorHelperData?: ExploreCorrelationHelperData;
+}
+const changeCorrelationHelperDataAction = createAction<ChangeCorrelationHelperData>(
+  'explore/changeCorrelationHelperData'
+);
+
+export function changeCorrelationHelperData(
+  exploreId: string,
+  correlationHelperData: ExploreCorrelationHelperData
+): ThunkResult<void> {
+  return async (dispatch, getState) => {
+    const exploreItem = getState().explore.panes[exploreId];
```

Comment:
Why do we need this check? When the action is dispatched, shouldn't it always come from an existing pane and exploreId should exist?